framework/include/userobjects/TimedElementSubdomainModifier.h

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ElementSubdomainModifier.h"
+#include "DelimitedFileReaderOfString.h"
+
+class TimedElementSubdomainModifier : public ElementSubdomainModifier
+{
+public:
+  static InputParameters validParams();
+
+  TimedElementSubdomainModifier(const InputParameters & parameters);
+
+  virtual void initialize();
+
+protected:
+  virtual SubdomainID computeSubdomainID() override;
+
+  virtual std::vector<real> onGetTimes();
+  virtual SubdomainID onComputeSubdomainID(real t_from_exclusive, real t_to_inclusive);
+
+  /// storage for the times including their original index.
+  struct timeIndexPair
+  {
+    real time;
+    std::size_t index;
+
+    bool operator<(const timeIndexPair & a) const
+    {
+      if (time == a.time)
+      {
+        return index < a.index;
+      }
+      else
+      {
+        return time < a.time;
+      };
+    }
+  };
+
+  /// Times and subdomain changes to make
+  std::vector<timeIndexPair> _times_and_indices;
+
+private:
+  /// Local variable for the current step (to be able to determine incrementation)
+  int _current_step;
+
+  /// Local variable for the end of the timespan to apply changes of the subdomain
+  real _current_t;
```

Comment:
```suggestion
  Real _current_t;
```
so by reference, I mean that you can declare this as
Real & _current_t

and set it with
```
_current_t(_fe_problem->time())
```

Reply:
I'm a bit unsure here: At the moment '_current_t' should always have the same value as '_t' *after* it gets updated in 'TimedElementSubdomainModifier::computeSubdomainID()' by '_current_t = _t;' along with '_last_t', and '_current_step'. 
During this process I have to get the correct value of '_last_t'. For this I use '_current_t'. If '_current_t' is defined by reference, dont I get a problem with '_last_t'?
Motivation: Up to my understanding the TimedElementSubdomainModifier does not control the points in time it gets called. So it can happen that the user had planned some block changes sometime between the last call and the current call (and not exactly at the time of the call.) I tried to design the TimedElementSubdomainModifier in such a way that these intermediate changes are then made up for.